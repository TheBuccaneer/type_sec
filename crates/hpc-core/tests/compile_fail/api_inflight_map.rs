// tests/compile_fail/api_inflight_map.rs
//! Compile-fail: Mapping ist aus InFlight nicht erlaubt.

#![deny(warnings)] // macht must_use etc. hart – nicht nötig, aber schadet nicht.

use hpc_core::*;

fn main() {
    // Wir brauchen keinen echten Wert; Typprüfung reicht aus, um den Fehler zu triggern.
    // Achtung: Wir erzeugen absichtlich einen "Phantom"-Wert, der nie benutzt wird.
    let q: Queue<'static> = unsafe { core::mem::zeroed() };
    let mut buf_inflight: DeviceBuffer<'static, u8, InFlight> = unsafe { core::mem::zeroed() };

    // Erwartet: **compile error** (kein Mapping aus InFlight erlaubt)
    // Falls deine API `map_write_blocking` heißt (wie in unseren Skizzen):
    let _mapped = buf_inflight
        .map_write_blocking(&q, 0..16, /*invalidate=*/ true)
        .unwrap();

    // Falls deine API stattdessen `map_for_write_block(...)` heißt,
    // kommentiere die Zeile oben aus und nutze diese hier:
    // let (_guard, _token) = buf_inflight.map_for_write_block(&q, 0..16).unwrap();
}
