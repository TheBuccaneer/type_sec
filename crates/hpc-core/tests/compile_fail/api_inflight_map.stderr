error[E0599]: no method named `map_write_blocking` found for struct `hpc_core::DeviceBuffer` in the current scope
  --> tests/compile_fail/api_inflight_map.rs:17:10
   |
16 |       let _mapped = buf_inflight
   |  ___________________-
17 | |         .map_write_blocking(&q, 0..16, /*invalidate=*/ true)
   | |         -^^^^^^^^^^^^^^^^^^ method not found in `hpc_core::DeviceBuffer<'static, u8, hpc_core::InFlight>`
   | |_________|
   |
